client/fingerprint/cpu.go

```diff
@@ -56,11 +60,25 @@ func (f *CPUFingerprint) Fingerprint(req *FingerprintRequest, resp *FingerprintR
 		f.logger.Debug("detected cpu frequency", "MHz", log.Fmt("%.0f", mhz))
 	}
 
-	if numCores := stats.CPUNumCores(); numCores > 0 {
+	var numCores int
+	if numCores = stats.CPUNumCores(); numCores > 0 {
 		resp.AddAttribute("cpu.numcores", fmt.Sprintf("%d", numCores))
 		f.logger.Debug("detected core count", "cores", numCores)
 	}
 
+	var reservableCores []uint16
+	if req.Config.ReservableCores != nil {
+		reservableCores = req.Config.ReservableCores
+		f.logger.Debug("reservable cores set by config", "cpuset", reservableCores)
+	} else {
+		if cores, err := f.deriveReservableCores(req); err != nil {
+			f.logger.Warn("failed to detect set of reservable cores", "error", err)
+		} else {
+			reservableCores = cpuset.New(cores...).Difference(cpuset.New(req.Node.ReservedResources.Cpu.ReservedCpuCores...)).ToSlice()
```

Comment:
this line is a bit gnarly, maybe worth splitting up. Is `req.Node.ReservedResources` guaranteed to not be nil?